### DISCUSSION TITLE
Run and save video or image through webcam or ip camera with using opencv python termux

### DISCUSSION THREAD

**Discussion by SamYokai** [termux/termux-packages#8456] [Ideas]
Want to ask, is that possible we use python termux to run and record a video or image through webcam or ip camera with using opencv? Or any suggestion other package that can run to capture video and save the video or images using python termux?

**Comment by ghost** [comment]
No, this is not possible. You can't access camera device through `/dev`. Even if can, there could be some proprietary protocol (Android doesn't use open source camera drivers). Also neither Termux or Termux:API expose socket or something like with video stream.

Pictures can be taken through Termux:API. You will need to run external command from your script with saving picture to a temporary file, then processing this file via OpenCV.

**Reply by SamYokai** [reply]
owh, I see. Because I just would like to try with video capture to capture the video and preprocess to image only. Thanks for the information.

**Comment by SDRausty** [comment]
>run and record a video or image through webcam
>other package that can run to capture video and save the video or images 

OpenCV with Python is superfluous.  Simple BASH can accomplish these tasks;  Hopefully these links will help:
a) [`cams`](https://github.com/shlibs/shlibs.bash/blob/master/termux-api/cams)
b) https://github.com/shlibs/shlibs.bash/commit/eaf328ae22df2c80a5f5b5c4b0a1622b021be517

**Reply by SamYokai** [reply]
If through BASH mean no need use Python is it? Then did this method can help to save the video capture and image? Sorry for my less experience and always ask...

**Reply by SDRausty** [reply]
>If through BASH mean no need use

Please read https://www.ecosia.org/search?q=BASH+scripting+FAQ+site%3Atldp.org then you should be able to use  [`cams`](https://github.com/shlibs/shlibs.bash/blob/master/termux-api/cams).
